controllers/src/main/java/com/redhat/ipaas/controllers/IntegrationController.java

```diff
@@ -125,7 +123,8 @@ private void checkIntegration(Integration integration) {
     private void enqueue(WorkflowHandler handler, String integrationId) {
         executor.submit(() -> {
             Integration integration = dataManager.fetch(Integration.class, integrationId);
-            if (stale(handler, integration)) {
+            String scheduledKey = "" + integration.getDesiredStatus() + ":" + integrationId;
```

Comment:
I was thinking that we mainly want to re-trigger the workflow processing immediately if a user changes the desired status.  Since current status is only changed by workflow actions, I think it's ok if those subsequent workflow actions occur after the scheduling delay.